EclEditor/LexSALT.cxx

```diff
@@ -0,0 +1,381 @@
+    WordList *[], Accessor &styler) {
+    bool foldComment = true;
+    bool foldPreprocessor = true;
+    bool foldCompact = true;
+    bool foldAtElse = true;
+    unsigned int endPos = startPos + length;
+    int visibleChars = 0;
+    int lineCurrent = styler.GetLine(startPos);
+    int levelCurrent = SC_FOLDLEVELBASE;
+    if (lineCurrent > 0)
+        levelCurrent = styler.LevelAt(lineCurrent - 1) >> 16;
+    int levelMinCurrent = levelCurrent;
+    int levelNext = levelCurrent;
+    char chNext = styler[startPos];
+    int styleNext = styler.StyleAt(startPos);
+    int style = initStyle;
+    for (unsigned int i = startPos; i < endPos; i++) {
+        char ch = chNext;
+        chNext = styler.SafeGetCharAt(i + 1);
+        int stylePrev = style;
+        style = styleNext;
+        styleNext = styler.StyleAt(i + 1);
+        bool atEOL = (ch == '\r' && chNext != '\n') || (ch == '\n');
+        if (foldComment && IsStreamCommentStyleSalt(style)) {
+            if (!IsStreamCommentStyleSalt(stylePrev)) {
+                levelNext++;
+            }
+            else if (!IsStreamCommentStyleSalt(styleNext) && !atEOL) {
+                // Comments don't end at end of line and the next character may be unstyled.
+                levelNext--;
+            }
+        }
+        if (foldComment && (style == SCE_SALT_COMMENTLINE)) {
+            if ((ch == '/') && (chNext == '/')) {
+                char chNext2 = styler.SafeGetCharAt(i + 2);
+                if (chNext2 == '{') {
+                    levelNext++;
+                }
+                else if (chNext2 == '}') {
+                    levelNext--;
+                }
+            }
+        }
+        if (atEOL || (i == endPos - 1)) {
+            int levelUse = levelCurrent;
+            if (foldAtElse) {
+                levelUse = levelMinCurrent;
+            }
+            int lev = levelUse | levelNext << 16;
+            if (visibleChars == 0 && foldCompact)
+                lev |= SC_FOLDLEVELWHITEFLAG;
+            if (levelUse < levelNext)
+                lev |= SC_FOLDLEVELHEADERFLAG;
+            if (lev != styler.LevelAt(lineCurrent)) {
+                styler.SetLevel(lineCurrent, lev);
+            }
+            lineCurrent++;
+            levelCurrent = levelNext;
+            levelMinCurrent = levelCurrent;
+            if (atEOL && (i == static_cast<unsigned int>(styler.Length() - 1))) {
+                // There is an empty line at end of file so give it same level and empty
+                styler.SetLevel(lineCurrent, (levelCurrent | levelCurrent << 16) | SC_FOLDLEVELWHITEFLAG);
+            }
+            visibleChars = 0;
+        }
+        if (!IsASpace(ch))
+            visibleChars++;
+    }
+}
+
+static const char * const SaltWordListDesc[] = {
+    "Commands",
+    "ParameterWithFields",
+    "Parameters",
+    "ParamsWithFieldlists",
+    "ParamsWithFieldCommas",
+    "SkipOneFieldlists",
+    0
+};
```

Comment:
These are the "titles" of the word lists used for coloring for all 6 types in the XML.